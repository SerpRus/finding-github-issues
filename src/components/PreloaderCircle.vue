<template>
  <div
    class="preloader"
    v-if="isLoader"
  >
  </div>
</template>

<script>
export default {
  name: 'PreloaderCircle',
  props: {
    isLoader: Boolean,
  },
};
</script>

<style lang="scss" scoped>
.preloader {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  display: flex;
  align-items: center;
  justify-content: center;

  &::before {
    display: block;
    width: 50px;
    height: 50px;

    content: '';
    animation: preloader-rotate 1s ease infinite;

    border: 8px solid #ccc;
    border-left: 8px solid brown;
    border-radius: 50%;

    @media (min-width: 767px) {
      width: 100px;
      height: 100px;

      border: 4px solid #ccc;
      border-left: 4px solid brown;
    }
  }
}

@keyframes preloader-rotate {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
</style>
