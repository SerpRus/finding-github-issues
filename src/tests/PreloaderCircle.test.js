import { mount } from '@vue/test-utils';
import PreloaderCircle from '../components/PreloaderCircle.vue';

describe('PreloaderCircle tests', () => {
  it('PreloaderCircle isLoader: true', () => {
    const wrapper = mount(PreloaderCircle, {
      props: {
        isLoader: true,
      },
    });

    expect(wrapper.html()).toBe('<div class="preloader"></div>');
  });

  it('PreloaderCircle isLoader: false', () => {
    const wrapper = mount(PreloaderCircle, {
      props: {
        isLoader: false,
      },
    });

    expect(wrapper.html()).toBe('<!--v-if-->');
  });

  it('PreloaderCircle isLoader: null', () => {
    const wrapper = mount(PreloaderCircle, {
      props: {
        isLoader: null,
      },
    });

    expect(wrapper.html()).toBe('<!--v-if-->');
  });

  it('PreloaderCircle isLoader: undefined', () => {
    const wrapper = mount(PreloaderCircle, {
      props: {
        isLoader: undefined,
      },
    });

    expect(wrapper.html()).toBe('<!--v-if-->');
  });

  it('PreloaderCircle no props', () => {
    const wrapper = mount(PreloaderCircle);

    expect(wrapper.html()).toBe('<!--v-if-->');
  });

  it('PreloaderCircle visible snapshot', () => {
    const wrapper = mount(PreloaderCircle, {
      props: {
        isLoader: true,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('PreloaderCircle hidden snapshot', () => {
    const wrapper = mount(PreloaderCircle, {
      props: {
        isLoader: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
