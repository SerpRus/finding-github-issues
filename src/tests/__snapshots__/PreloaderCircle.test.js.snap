// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreloaderCircle tests PreloaderCircle hidden snapshot 1`] = `"<!--v-if-->"`;

exports[`PreloaderCircle tests PreloaderCircle visible snapshot 1`] = `"<div class=\\"preloader\\"></div>"`;
